homeassistant/components/sensor/linux_battery.py

```diff
@@ -35,45 +35,57 @@
 ATTR_VOLTAGE_MIN_DESIGN = 'voltage_min_design'
 ATTR_VOLTAGE_NOW = 'voltage_now'
 
+ATTR_HEALTH = 'health'
+ATTR_PRESENT = 'present'
+ATTR_STATUS = 'status'
+
 CONF_BATTERY = 'battery'
+CONF_DISTRIBUTION = 'distribution'
 
 DEFAULT_BATTERY = 1
-DEFAULT_NAME = 'Battery'
+DEFAULT_NAME = 'battery'
 DEFAULT_PATH = '/sys/class/power_supply'
+DEFAULT_DISTRIBUTION = 'linux'
 
 ICON = 'mdi:battery'
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
     vol.Optional(CONF_BATTERY, default=DEFAULT_BATTERY): cv.positive_int,
     vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_DISTRIBUTION, default=DEFAULT_DISTRIBUTION): cv.string,
 })
 
 
 def setup_platform(hass, config, add_devices, discovery_info=None):
     """Set up the Linux Battery sensor."""
     name = config.get(CONF_NAME)
     battery_id = config.get(CONF_BATTERY)
+    linux_distribution = config.get(CONF_DISTRIBUTION)
 
     try:
-        os.listdir(os.path.join(DEFAULT_PATH, 'BAT{}'.format(battery_id)))
+        if linux_distribution.lower() == 'android':
+            os.listdir(os.path.join(DEFAULT_PATH, name))
```

Comment:
This will not work as `CONF_NAME` is the friendly for the frontend and the user can change that.